release/pkg/filereader/file_reader.go

```diff
@@ -178,88 +178,93 @@ func GetEksDReleaseManifestUrl(releaseChannel, releaseNumber string, dev bool) s
+// Get the build number for the current eks-a dev build
+func GetCurrentEksADevBuildNumber(releaseVersion string, r *releasetypes.ReleaseConfig) (int, error) {
+	tempFileName := "latest-dev-release-version"
 
-	var newDevReleaseVersion string
-	if r.Weekly {
-		newDevReleaseVersion = fmt.Sprintf("v0.0.0-dev+build.%s", r.ReleaseDate)
+	var latestReleaseKey, latestBuildVersion string
+	var currentEksaBuildNumber int
+	if r.BuildRepoBranchName == "main" {
+		latestReleaseKey = "LATEST_RELEASE_VERSION"
 	} else {
-		tempFileName := "latest-dev-release-version"
-
-		var latestReleaseKey string
-		if r.BuildRepoBranchName == "main" {
-			latestReleaseKey = "LATEST_RELEASE_VERSION"
-		} else {
-			latestReleaseKey = fmt.Sprintf("%s/LATEST_RELEASE_VERSION", r.BuildRepoBranchName)
+		latestReleaseKey = fmt.Sprintf("%s/LATEST_RELEASE_VERSION", r.BuildRepoBranchName)
+	}
+	if s3.KeyExists(r.ReleaseBucket, latestReleaseKey) {
+		err := s3.DownloadFile(tempFileName, r.ReleaseBucket, latestReleaseKey)
+		if err != nil {
+			return -1, errors.Cause(err)
+		}
+		// Check if current version and latest version are the same semver
+		latestBuildS3, err := os.ReadFile(tempFileName)
+		if err != nil {
+			return -1, errors.Cause(err)
+		}
+		latestBuildVersion = string(latestBuildS3)
+		if releaseVersion == "vDev" { // TODO: remove when we update the pipeline
+			releaseVersion = "v0.0.0"
 		}
-		if s3.KeyExists(r.ReleaseBucket, latestReleaseKey) {
-			err := s3.DownloadFile(tempFileName, r.ReleaseBucket, latestReleaseKey)
-			if err != nil {
-				return "", errors.Cause(err)
-			}
-
-			// Check if current version and latest version are the same semver
-			latestBuildS3, err := os.ReadFile(tempFileName)
-			if err != nil {
-				return "", errors.Cause(err)
-			}
-			latestBuildVersion := string(latestBuildS3)
-			newDevReleaseVersion, err = GenerateNewDevReleaseVersion(latestBuildVersion, releaseVersion, r.BuildRepoBranchName)
-			if err != nil {
-				return "", errors.Cause(err)
-			}
-			fmt.Printf("Previous release version: %s\n", latestBuildVersion)
-			fmt.Printf("Current release version provided: %s\n", releaseVersion)
-		} else {
-			newDevReleaseVersion = "v0.0.0-dev+build.0"
-			if r.BuildRepoBranchName != "main" {
-				newDevReleaseVersion = fmt.Sprintf("v0.0.0-dev-%s+build.0", r.BuildRepoBranchName)
-			}
+		currentEksaBuildNumber, err = GetNewEksaDevBuildNumber(latestBuildVersion, releaseVersion, r.BuildRepoBranchName)
+		if err != nil {
+			return -1, errors.Cause(err)
 		}
+	} else {
+		currentEksaBuildNumber = 0
 	}
-
-	fmt.Printf("New dev release release version: %s\n", newDevReleaseVersion)
-
-	fmt.Printf("%s Successfully computed current dev release version\n", constants.SuccessIcon)
-	return newDevReleaseVersion, nil
+	return currentEksaBuildNumber, nil
 }
 
-func GenerateNewDevReleaseVersion(latestBuildVersion, releaseVersion, branchName string) (string, error) {
+// Updates the last dev build number by 1
```

Comment:
this comment is a bit misleading
It doesn't really update anything, it just bump the current number by one if found
Also doc comments are usually like
```suggestion
// GetNewEksaDevBuildNumber reads bklablab and calculates blblalb,
```